### PR TITLE
[Perpetual] net open interest was added

### PR DESCRIPTION
**Pull request by fenriz07**
Net open interest was added to the next queries

- show-pool
- list-pool

### CODE REVIEW

File: x/perpetual/keeper/begin_blocker.go [hunk 1]
```diff
@@ -66,7 +66,13 @@ func (k Keeper) BeginBlocker(ctx sdk.Context) {
 				FundingRateLong:    fundingRateLong,
 				FundingRateShort:   fundingRateShort,
 			})
+
+			err = k.UpdateNetOpenInterest(ctx, &pool)
```
Comment (cosmic-vagabond) [inline x/perpetual/keeper/begin_blocker.go:70]: @fenriz07 I would refrain from adding any unnecessary computation at the ABCI level (begin/end blocker), can you instead do this computation within the query function, another option would be to calculate that value within an offchain indexer. cc @Wgil @amityadav0 @avkr003

Comment (amityadav0) [reply x/perpetual/keeper/begin_blocker.go:70]: agree, we can add this in query, also we don;t need to store it as well. So no migration is needed

File: Makefile [hunk 2]
```diff
@@ -4,7 +4,7 @@
 NAME?=elys
 BINARY?=$(NAME)d
 COMMIT:=$(shell git log -1 --format='%H')
-VERSION:=$(shell git describe --tags --match 'v*' --abbrev=8 | sed 's/-g/-/' | sed 's/-[0-9]*-/-/')
+VERSION:=v999.999.999
```
Comment (cosmic-vagabond) [inline Makefile:7]: @fenriz07 why did you change that?

**Comment by cosmic-vagabond** [comment]
@fenriz07 can you share a test result in comment of the updated CLI query thanks

**Comment by fenriz07** [comment]
elysd q perpetual list-pool --output=json 

`{"pool":[{"amm_pool_id":"2","health":"0.883351042402733204","enabled":true,"closed":false,"borrow_interest_rate":"0.000000535466554908","pool_assets_long":[{"liabilities":"3931813932907","custody":"0","take_profit_liabilities":"3605857570107","take_profit_custody":"0","asset_balance":"1062716517114","block_borrow_interest":"0","asset_denom":"ibc/2180E84E20F5679FCC760D8C165B60F42065DEF7F46A72B447CFF1B7DC6C0A65"},{"liabilities":"0","custody":"563524739564","take_profit_liabilities":"0","take_profit_custody":"394720462935","asset_balance":"-563584155567","block_borrow_interest":"0","asset_denom":"ibc/E2D2F6ADCC68AA3384B2F5DFACCA437923D137C14E86FB8A10207CF3BED0C8D4"}],"pool_assets_short":[{"liabilities":"0","custody":"25521549362","take_profit_liabilities":"0","take_profit_custody":"758800649","asset_balance":"-13515873900","block_borrow_interest":"0","asset_denom":"ibc/2180E84E20F5679FCC760D8C165B60F42065DEF7F46A72B447CFF1B7DC6C0A65"},{"liabilities":"3173650598","custody":"0","take_profit_liabilities":"758578518","take_profit_custody":"0","asset_balance":"0","block_borrow_interest":"0","asset_denom":"ibc/E2D2F6ADCC68AA3384B2F5DFACCA437923D137C14E86FB8A10207CF3BED0C8D4"}],"last_height_borrow_interest_rate_computed":"9865471","funding_rate":"0.001000000000000000","net_open_interest":"3928640282309"},{"amm_pool_id":"4","health":"1.000010318825890871","enabled":true,"closed":false,"borrow_interest_rate":"0.000000219967299502","pool_assets_long":[{"liabilities":"-200000000","custody":"0","take_profit_liabilities":"-186115695","take_profit_custody":"0","asset_balance":"6781429","block_borrow_interest":"0","asset_denom":"ibc/2180E84E20F5679FCC760D8C165B60F42065DEF7F46A72B447CFF1B7DC6C0A65"},{"liabilities":"0","custody":"0","take_profit_liabilities":"0","take_profit_custody":"-445632798","asset_balance":"0","block_borrow_interest":"0","asset_denom":"ibc/B4314D0E670CB43C88A5DCA09F76E5E812BD831CC2FEC6E434C9E5A9D1F57953"}],"pool_assets_short":[{"liabilities":"0","custody":"0","take_profit_liabilities":"0","take_profit_custody":"0","asset_balance":"320805","block_borrow_interest":"0","asset_denom":"ibc/2180E84E20F5679FCC760D8C165B60F42065DEF7F46A72B447CFF1B7DC6C0A65"},{"liabilities":"0","custody":"0","take_profit_liabilities":"-1","take_profit_custody":"0","asset_balance":"0","block_borrow_interest":"0","asset_denom":"ibc/B4314D0E670CB43C88A5DCA09F76E5E812BD831CC2FEC6E434C9E5A9D1F57953"}],"last_height_borrow_interest_rate_computed":"9865471","funding_rate":"0.000000000000000000","net_open_interest":"-200000000"}],"pagination":{"next_key":null,"total":"0"}}`

**Comment by fenriz07** [comment]
elysd q perpetual show-pool 2 --output=json

`{"pool":{"amm_pool_id":"2","health":"0.883351042402733204","enabled":true,"closed":false,"borrow_interest_rate":"0.000000535466554908","pool_assets_long":[{"liabilities":"3931813932907","custody":"0","take_profit_liabilities":"3605857570107","take_profit_custody":"0","asset_balance":"1062716517114","block_borrow_interest":"0","asset_denom":"ibc/2180E84E20F5679FCC760D8C165B60F42065DEF7F46A72B447CFF1B7DC6C0A65"},{"liabilities":"0","custody":"563524739564","take_profit_liabilities":"0","take_profit_custody":"394720462935","asset_balance":"-563584155567","block_borrow_interest":"0","asset_denom":"ibc/E2D2F6ADCC68AA3384B2F5DFACCA437923D137C14E86FB8A10207CF3BED0C8D4"}],"pool_assets_short":[{"liabilities":"0","custody":"25521549362","take_profit_liabilities":"0","take_profit_custody":"758800649","asset_balance":"-13515873900","block_borrow_interest":"0","asset_denom":"ibc/2180E84E20F5679FCC760D8C165B60F42065DEF7F46A72B447CFF1B7DC6C0A65"},{"liabilities":"3173650598","custody":"0","take_profit_liabilities":"758578518","take_profit_custody":"0","asset_balance":"0","block_borrow_interest":"0","asset_denom":"ibc/E2D2F6ADCC68AA3384B2F5DFACCA437923D137C14E86FB8A10207CF3BED0C8D4"}],"last_height_borrow_interest_rate_computed":"9865494","funding_rate":"0.001000000000000000","net_open_interest":"3928640282309"}}
`

@cosmic-vagabond

File: x/perpetual/keeper/get_net_open_interest.go [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+	for _, asset := range pool.PoolAssetsLong {
+		assetLiabilitiesLong = assetLiabilitiesLong.Add(asset.Liabilities)
+	}
+
+	for _, asset := range pool.PoolAssetsShort {
+		assetLiabilitiesShort = assetLiabilitiesShort.Add(asset.Liabilities)
+	}
+
+	netOpenInterest := assetLiabilitiesLong.Sub(assetLiabilitiesShort)
```
Comment (cosmic-vagabond) [inline x/perpetual/keeper/get_net_open_interest.go:21]: @fenriz07 there is one thing that is not right and I realized we are doing it wrong with the existing `update funding rate` function here https://github.com/elys-network/elys/blob/main/x/perpetual/keeper/update_funding_rate.go as well, we have to make sure that all liabilities amount that we are cumulating within `assetLiabilitiesLong` and `assetLiabilitiesShort` are using the same denom `USDC` if not they must be swapped to USDC the same way we do here https://github.com/elys-network/elys/blob/main/x/perpetual/keeper/keeper.go
can you update both `get net open interest` and `update funding rate` functions to reflect that? thanks

Comment (cosmic-vagabond) [reply x/perpetual/keeper/get_net_open_interest.go:21]: this issue mentioned above will be handle as part of a separate PR, everything else looks good

**Review by cosmic-vagabond — APPROVED** [review]
